getting-started/deploying-to-openshift.adoc

```diff
@@ -0,0 +1,235 @@
+## Route TLS configuration:
+quarkus.openshift.route.tls.termination=edge
+quarkus.openshift.route.tls.insecure-edge-termination-policy=None
+----
+
+. **Build and deploy the application:**
++
+[source,shell]
+----
+mvn install -Dquarkus.openshift.deploy=true
+----
+
+=== 2. Build Locally with Docker and Automatically Deploy on OpenShift
+
+This method involves building the container image for the application locally using Docker, pushing the image to registry, followed by an automatic deployment to OpenShift.
+
+[.steps]
+
+. **Log in to OpenShift CLI:**
++
+[source,shell]
+----
+oc login
+----
+
+. **Add the OpenShift extension to the Quarkus project:**
++
+[source,shell]
+----
+mvn quarkus:add-extension -Dextensions='quarkus-openshift'
+----
+
+. **Update the `src/main/resources/application.properties` file with the following properties to configure the image and enable the creation of an OpenShift Route:**
++
+[source,properties]
+----
+# Container Image Builder
+quarkus.container-image.builder=docker
+quarkus.container-image.registry=<image_registry>
+quarkus.container-image.group=<registry_group_or_user>
+quarkus.container-image.name=<image_name>
+quarkus.container-image.tag=<image_tag>
+quarkus.docker.buildx.platform=linux/amd64
+# OpenShift
+quarkus.openshift.route.expose=true
+## Route TLS configuration:
+quarkus.openshift.route.tls.termination=edge
+quarkus.openshift.route.tls.insecure-edge-termination-policy=None
+----
+
+[NOTE]
+====
+* `<image_registry>`: The URL of the container image registry where the Docker image will be stored. Example: `docker.io`.
+* `<registry_group_or_user>`: The group or user in the container image registry. Example: `myuser`.
+* `<image_name>`: The name of the Docker image to be built. Example: `myapp`.
+* `<image_tag>`: The tag of the Docker image. Example: `v1.0.0`.
+
+Replace these tokens with appropriate values based on the environment setup.
+====
+
+. **Add the `Dockerfile.jvm` file to `src/main/docker`:**
++
+[source,Dockerfile]
+----
+FROM icr.io/appcafe/ibm-semeru-runtimes:open-17-jre-ubi-minimal
+
+# Switch to root
+USER 0
+
+# Copy the compiled application files from the local `target` directory to the `/app/` directory inside the container.
+COPY target /app/
+
+# Support Arbitrary User IDs
+# https://docs.openshift.com/container-platform/3.11/creating_images/guidelines.html#openshift-specific-guidelines
+RUN chgrp -R 0 /app && \
+  chmod -R g+rwX /app
+
+# Switch back to a non-root user
+USER 1001
+
+WORKDIR /app
+
+EXPOSE 8080
+
+CMD ["java", "-jar", "quarkus-app/quarkus-run.jar"]
+----
+
+. **Build and deploy the application:**
++
+[source,shell]
+----
+mvn install -Dquarkus.openshift.deploy=true
+----
+
+=== 3. Build Locally with Docker and Create Manual Deployment on OpenShift
+
+This option provides a way to build the the container image for the application locally using Docker, push the image to a registry, and manually deploy it on OpenShift.
+
+[.steps]
+
```

Comment:
Shouldn't need this extra blank line

Reply:
Do we have any kind of automatic formatting configured for this repo?

Reply:
No, there's no asciidoc linter or formatter that I'm aware of.

Reply:
I can try to implement this [1] into a GitHub action for this repo, publishing the docs to a GH Page, in a subpath per branch. Should be easy enough.

[1] https://github.com/marketplace/actions/asciidoctor-ghpages

Reply:
Not sure what that would accomplish for us. That just builds the adoc so you can publish the html. GitHub will render it in the PR, or when you browse it.

Reply:
Oh, you're right!